client.go

```diff
@@ -65,7 +68,13 @@ type Client struct {
 // submission blocks on the first attempt. When connection establishment fails,
 // then command submissions receive the error of the last attempt, until the
 // connection restores. A zero connectTimeout defaults to one second.
-func NewClient(addr string, commandTimeout, connectTimeout time.Duration) *Client {
+func NewClient(addr string, commandTimeout, connectTimeout time.Duration, password ...string) *Client {
+	return NewClientWithAuth(addr, commandTimeout, connectTimeout, "")
+}
+
+// NewClientWithAuth launches a managed connection to a service
+// address just like NewClient but also uses password authentication.
+func NewClientWithAuth(addr string, commandTimeout, connectTimeout time.Duration, password string) *Client {
```

Comment:
We could use the same approach for AUTH as done with SELECT now. You'll need something like a redis.Client.password atomic.Value though.

Alternatively we could provide a more generic redis.NewClientWithConfig(redis.Config) *redis.Client next to the existing redis.NewClient.

```go
type Config struct {
        Addr           string
        CommandTimeout time.Duration
        ConnectTimeout time.Duration
        Password       string
}
```

Reply:
Maybe SELECT was a mistake, and we should have a redis.Config.DB int64 instead.

Reply:
What do you think @squeaky2x3?